uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/models/work-pattern-model_test.js

```diff
@@ -0,0 +1,125 @@
+/* eslint-env amd, jasmine */
+/* global inject */
+
+define([
+  'common/lodash',
+  'leave-absences/shared/models/work-pattern-model',
+  'mocks/apis/work-pattern-api-mock'
+], function (_) {
+  'use strict';
+
+  describe('WorkPattern', function () {
+    var $provide, WorkPattern, WorkPatternAPI, $rootScope;
+
+    beforeEach(module('leave-absences.models', 'leave-absences.mocks', 'common.mocks',
+      function (_$provide_) {
+        $provide = _$provide_;
+      }));
+
+    beforeEach(inject(function (_WorkPatternAPIMock_) {
+      $provide.value('WorkPatternAPI', _WorkPatternAPIMock_);
+    }));
+
+    beforeEach(inject(function (_WorkPattern_, _WorkPatternAPI_, _$rootScope_) {
+      WorkPattern = _WorkPattern_;
+      WorkPatternAPI = _WorkPatternAPI_;
+      $rootScope = _$rootScope_;
+
+      spyOn(WorkPatternAPI, 'get').and.callThrough();
+      spyOn(WorkPatternAPI, 'workPatternsOf').and.callThrough();
+      spyOn(WorkPatternAPI, 'assignWorkPattern').and.callThrough();
+    }));
+
+    it('has expected interface', function () {
+      expect(Object.keys(WorkPattern)).toEqual(['assignWorkPattern', 'default', 'workPatternsOf']);
+    });
+
+    describe('basic tests', function () {
```

Comment:
This was done because there was a unit test which did not need `$rootScope.$apply()`. As discussed removed the `describe` block.